jupyterhub/templates/ingress.yaml

```
@@ -24,6 +24,19 @@ spec:
                 name: {{ include "jupyterhub.proxy-public.fullname" $ }}
                 port:
                   name: http
+          {{- range $path := $.Values.ingress.extraPaths }}
+          - path: {{ $path.path }}
+            pathType: {{ $path.pathType }}
+            backend:
+              service:
+                name: {{ $path.backend.service.name }}
+                port:
+                  {{- if $path.backend.service.port.number }}
+                  number: {{ $path.backend.service.port.number }}
+                  {{- else if $path.backend.service.port.name }}
+                  name: {{ $path.backend.service.port.name }}
+                  {{- end }}
+          {{- end }}
```

Does this work as well? It would allow us to not need to keep updated with all options we can specify, for example having `backend.resource` which is valid as well.

```suggestion
          {{- with $.Values.ingress.extraPaths }}
          {{- . | toYaml | nindent 10 }}
          {{- end }}
```

You could trial this a bit by adding an entry for `ingress.extraPaths` in the lint-and-validate-config.yaml file, which also helps us test this a bit.

---

it's even better indeed, i'll test this